### PR TITLE
Extract deployment from transformation process

### LINKED ISSUE #408: Extract deployment logic of the transformation
To push files to the CF instance while the transformation process is not a good way because we hide this from the user and create e.g. services without his knowing.
To avoid this we have to extract this logic into the deployment-scripts. Consequently, we have to do a bigger rework to the java-implementation. 
- [x] no deploying while transformation
- [x] provide scripts for readingCredentials and so on
- [x] add code to bash script. 

EDIT: "Implement logic to transformation" I will create a new PR for this because it might be bigger changes.

A first approach of the used scripts you could find [here](https://hackmd.io/s/B14EwBL4f#)

### PR DESCRIPTION
**Pull request by JensMueller2709**
resolves #408 , #375 , #395

### CODE REVIEW

File: server/src/main/java/org/opentosca/toscana/plugins/cloudfoundry/application/deployment/Deployment.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    private final String PYTHON_EXECUTE_FILE = "/cloudFoundry/deployment_scripts/executeCommand.py";
+    private final String PYTHON_EXECUTE_FILE_TARGET = "scripts/executeCommand.py";
+    private final String PYTHON_READ_CREDENTIALS = "/cloudFoundry/deployment_scripts/readCredentials.py";
+    private final String PYTHON_REPLACE_STRINGS = "/cloudFoundry/deployment_scripts/replace.py";
+
+    public Deployment(BashScript deploymentScript, Application application, PluginFileAccess fileAccess) throws IOException {
+        this.deploymentScript = deploymentScript;
+        this.application = application;
+        this.fileAccess = fileAccess;
+        clazz = Deployment.class;
```
Comment (orferch) [inline server/src/main/java/org/opentosca/toscana/plugins/cloudfoundry/application/deployment/Deployment.java]: rename clazz

File: server/src/main/java/org/opentosca/toscana/plugins/cloudfoundry/client/InjectionHandler.java [hunk 2]
```diff
@@ -48,7 +50,8 @@ public boolean deploy() {
         boolean succeed = false;
         try {
             if (connection != null) {
-                Path pathToApplication = Paths.get(fileAccess.getAbsolutePath(app.getPathToApplication()));
+                Path pathToApplication = Paths.get(fileAccess.getAbsolutePath(APPLICATION_FOLDER +
```
Comment (orferch) [inline server/src/main/java/org/opentosca/toscana/plugins/cloudfoundry/client/InjectionHandler.java]: Stringbuilder would be better here

**Comment by JensMueller2709** [comment]
this PR is still on WIP, so I am working on it. But of course I will consider your requested changes

File: docs/dev/plugins/cloudFoundry/Script-Overview.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+## Order
+1. `cf create service $servicename $serviceplan $serviceInstanceName`
+2. `cf push $appName -f $pathToManifest`
+3. `python readCredentials.py $appName $serviceName $serviceType`
+4. `python replace.py $file $findString $replaceString`
+5. `python executeScript.py $pathInWardenContainter $appName`
+6. `python configureMysql.py $sqlFile` or 12Factor: `python executeScript.py $appName $dbInitScript`
+
+## Summary
+the AWS Cloudfourmation CLI needs python anyways so we could use it too. We think for "bigger" script implementation python is the best way to do it.
```
Comment (mklopp) [inline docs/dev/plugins/cloudFoundry/Script-Overview.md]: Cloudformation

File: server/src/main/java/org/opentosca/toscana/plugins/cloudfoundry/application/deployment/Deployment.java [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+public class Deployment {
+
+    private BashScript deploymentScript;
+    private Application application;
+    private PluginFileAccess fileAccess;
+    private Class deploymentClass;
+    private boolean pythonIsChecked;
+
+    private final String PYTHON_SCRIPTS_TARGET = SCRIPTS_DIR_PATH;
```
Comment (mklopp) [inline server/src/main/java/org/opentosca/toscana/plugins/cloudfoundry/application/deployment/Deployment.java]: These should be static?

File: server/src/main/java/org/opentosca/toscana/plugins/cloudfoundry/application/deployment/Deployment.java [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    private boolean isAlreadyCopied(String fileName) {
+        try {
+            fileAccess.getAbsolutePath(PYTHON_SCRIPTS_TARGET + fileName);
+        } catch (FileNotFoundException e) {
+            return false;
+        }
+        return true;
+    }
+
+    private void checkPython() throws IOException {
```
Comment (mklopp) [inline server/src/main/java/org/opentosca/toscana/plugins/cloudfoundry/application/deployment/Deployment.java]: You do this `checkPyhton()` after every call to `copyFile()` so you can just include it that method

File: server/src/main/resources/cloudFoundry/deployment_scripts/configureMysql.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+import ast
+import sys
+
+# needs a special python mysql connector installation
+def main():
+    mysqlConfigFile = 'mysqlConfig.txt'
+    # should be a sql file
+    strConfigureFile = str(sys.argv[1])
+
+    if ".sql" not in strConfigureFile[len(strConfigureFile)-4:]:
```
Comment (mklopp) [inline server/src/main/resources/cloudFoundry/deployment_scripts/configureMysql.py]: consider using: `".sql" == os.path.splitext(strConfigureFile)[-1].lower()` or `strConfigureFile.lower().endswith(".sql")`

File: server/src/main/resources/cloudFoundry/deployment_scripts/configureMysql.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        print("needs an .sql File for db init")
+        exit(1)
+
+    try:
+        with open(strConfigureFile, 'r') as dbinit_file:
+            dbinit_command = dbinit_file.read()
+    except IOError as err:
+        print("Failed to read dbinit File named %s. Where it is?" %(strConfigureFile))
```
Comment (mklopp) [inline server/src/main/resources/cloudFoundry/deployment_scripts/configureMysql.py]: consider using `print("Failed to read dbinit File named {}. Where it is?".format(strConfigureFile))`

File: server/src/main/resources/cloudFoundry/deployment_scripts/configureMysql.py [hunk 8]
```diff
@@ -0,0 +1,47 @@
+import mysql.connector
```
Comment (mklopp) [inline server/src/main/resources/cloudFoundry/deployment_scripts/configureMysql.py]: how do you know this package is installed?

File: server/src/main/java/org/opentosca/toscana/plugins/cloudfoundry/application/deployment/Deployment.java (hunk 5 above)
Comment (JensMueller2709) [reply server/src/main/java/org/opentosca/toscana/plugins/cloudfoundry/application/deployment/Deployment.java]: Done

**Pull request merged**